### PR TITLE
Feature/xml standard [4115]

### PR DESCRIPTION
**Pull request by coquisx**
Adapted xmlparser to standard DDSXTypes

### CODE REVIEW

File: src/cpp/xmlparser/XMLParser.cpp [hunk 1]
```diff
@@ -881,15 +913,176 @@ p_dynamictypebuilder_t XMLParser::parseXMLMemberDynamicType(tinyxml2::XMLElement
...
+            types::DynamicTypeBuilder* innerBuilder =
+                    factory->CreateMapBuilder(keyTypeBuilder, valueTypeBuilder, length);
+            std::vector<uint32_t> bounds;
+            dimensionsToArrayBounds(memberArray, bounds);
+            memberBuilder = factory->CreateArrayBuilder(innerBuilder, bounds);
+            //factory->DeleteBuilder(innerBuilder);
+        }
+    }
+
+//TODO add arrays
```
Comment (LuisGP) [inline src/cpp/xmlparser/XMLParser.cpp]: Arrays must be implemented before merge.

File: src/cpp/xmlparser/XMLParser.cpp [hunk 2]
```diff
@@ -881,15 +913,176 @@ p_dynamictypebuilder_t XMLParser::parseXMLMemberDynamicType(tinyxml2::XMLElement
...
+        {
+            logError(XMLPARSER, "Error parsing sequence element type: Cannot be recognized.");
+            return nullptr;
+        }
+
+        const char* lengthStr = p_root->Attribute(SEQ_MAXLENGTH);
+        uint32_t length = MAX_ELEMENTS_COUNT;
+        if (lengthStr != nullptr)
+        {
+            length = std::stoi(lengthStr);
```
Comment (LuisGP) [inline src/cpp/xmlparser/XMLParser.cpp]: Be careful, `lengthStr` could include expressions.

File: src/cpp/xmlparser/XMLParser.cpp [hunk 3]
```diff
@@ -881,15 +913,176 @@ p_dynamictypebuilder_t XMLParser::parseXMLMemberDynamicType(tinyxml2::XMLElement
...
+        {
+            logError(XMLPARSER, "Error parsing map's value element type: Cannot be recognized.");
+            return nullptr;
+        }
+
+        const char* lengthStr = p_root->Attribute(MAP_MAXLENGTH);
+        uint32_t length = MAX_ELEMENTS_COUNT;
+        if (lengthStr != nullptr)
+        {
+            length = std::stoi(lengthStr);
```
Comment (LuisGP) [inline src/cpp/xmlparser/XMLParser.cpp]: Be careful, `lengthStr` could include expressions.

File: src/cpp/xmlparser/XMLParser.cpp (hunk 1 above)
Comment (coquisx) [reply src/cpp/xmlparser/XMLParser.cpp]: Nothing to do

File: src/cpp/xmlparser/XMLParser.cpp [hunk 4]
```diff
@@ -612,6 +612,9 @@ static p_dynamictypebuilder_t getDiscriminatorTypeBuilder(const std::string &dis
     {
         return factory->CreateChar16Builder();
     }
+
+    //TODO add CreateString
```
[the remark this answers is not in the document]
Comment (coquisx) [reply src/cpp/xmlparser/XMLParser.cpp]: Done

File: src/cpp/xmlparser/XMLParser.cpp (hunk 2 above)
Comment (coquisx) [reply src/cpp/xmlparser/XMLParser.cpp]: For future features

File: src/cpp/xmlparser/XMLParser.cpp (hunk 3 above)
Comment (coquisx) [reply src/cpp/xmlparser/XMLParser.cpp]: For future features

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1062/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1032/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1046/)

**Pull request merged**